zipline/assets/assets.py

```diff
@@ -648,9 +648,10 @@ def _lookup_symbol_fuzzy(self, symbol, as_of_date):
 
         if not options:
             # no equity owned the fuzzy symbol on the date requested
-            SymbolNotFound(symbol=symbol)
+            raise SymbolNotFound(symbol=symbol)
+
 
-        if len(options) == 1:
+        if len(options) == 1 or len(set(map(itemgetter(0), options))) == 1:
```

Comment:
I notice that we're maintaining these option pairs in a list, but then here and potentially later on, we dedup them on sid.  Would it make sense to use an `OrderedDict` instead of the list, or is that much slower in most cases?